### ISSUE TITLE
SCA: Saving a card at checkout can fail on extreme situations

### ISSUE THREAD

**Issue by DanReyLop** [woocommerce/woocommerce-square#3]
To reproduce:
- Make sure you're using a EU (or UK) Square Sandbox account.
- Buy an item in the store, go to checkout.
- Use an EU card from here: https://developer.squareup.com/docs/testing/test-values#test-values
- You'll get the SCA verification modal.
- The payment will fail with a `CARD_DECLINED_VERIFICATION_REQUIRED`, even if you provide the correct verification code. The card will be saved correctly though.

There's more context about this issue here: https://github.com/woocommerce/woocommerce-square-private/pull/270#issuecomment-539237443

Basically, Square's test cards **always** require SCA authentication. Because saving a card and paying with it are 2 separate operations, Square will require authentication twice, even if those operations are separated by a mere few seconds.

This is unlikely to become a problem in real-life situations (banks won't require authentication so aggressively), that's why this issue is low priority. However, ideally the extension working shouldn't be dependent on how aggressive the bank is.

**Comment by imodouglas** [comment]
7637919-zen

**Comment by jeffpaul** [comment]
Leaving this as a low priority as this appears to only affect Sandbox accounts and not Production accounts (thus not impacting "live" transactions).  If someone is able to reproduce with a Production account, then we can investigate further.

**Comment by spraveenitpro** [comment]
> Leaving this as a low priority as this appears to only affect Sandbox accounts and not Production accounts (thus not impacting "live" transactions). If someone is able to reproduce with a Production account, then we can investigate further.

Hi @jeffpaul , The customer has confirmed that this is happening for both test and live transactions 7893020-zen

**Comment by jeffpaul** [comment]
@spraveenitpro have you confirmed that the reproduction steps noted in the original issue description match the steps for live transactions?

**Comment by wyter** [comment]
@jeffpaul, I've asked the merchant to confirm this. We'll inform you.

**Comment by spraveenitpro** [comment]
@jeffpaul just heard back from customer, This is on initial payment, no card details have been saved.

**Comment by qasumitbagthariya** [comment]
I attempted to verify the issue using a sandbox account and encountered the same problem.

To reproduce the issue, try setting up a UK store and follow the steps to regenerate the error.

- WordPress: 6.7.2
- Theme: Twenty Twenty-Four 1.3
- WooCommerce - 9.6.2
- PHP: 8.0.30
- Web Server: Nginx 1.20.2
- Browser: Chrome 
- OS: macOS 15.2 
- Branch: trunk

https://github.com/user-attachments/assets/7e2aac3e-baeb-4a31-8907-94fae50f7133

cc @vikrampm1
